### DISCUSSION TITLE
DnD modding or similar

### DISCUSSION THREAD

**Discussion by soge-king** [saertna/obsidian-gamified-pkm#60] [Ideas]
im curious as to how to edit/modify the setting so that i can sort of do a similar Statblock like DnD(INT,CON,STR,DEX,CHAR,WIS), and each task would give me xp/lvl up an "ability"/make daily tasks that improve a stat/etc. i also want to try and integrate the Rewarded aspect. like lvling up or compleating a task gives you a common/rare/epic/legendary reward (have a snack/take a nap/play a agme etc)

Also, just some ideas.  
1 - Preset List of lvl names related to chosen class (class is arbitrary, but does give a better feel for progression. ill show what i added to my avatar page)  
2- Relate Specific folder and/or notes to a certain stat or skill. the xp for a stat upgrade should be a lot higher and correlate with the skill asociated with it (exp.:  
Task/Note/File ->Skill/Ability -> STAT  
Cardio (in daily notes) -> Sprinting -> CON  
Updating a note on a math subject -> Mathematics -> INT)  
3 - set maturaty lvls to a selected number/ this can correlate with how hard/easy a task or subject is  
4 - select a photo/icon to represent the "player" i.e. you  
5 - a selectable sound byte per completed tasks/lvl up. this i just like because it feels nice to hear a "ding" when you finish some. could be my ADHD, but whatever

so i like artificers and i went with that theme  so the lvls would be like:
level 5: *Novice Artificer*
level 10: *Experienced Initiate*
level 20: *Skilled Artificer*
level 30: *Expert Artificer*
level 40: *Mechanist Initiate*
level 50: *Skilled Mechanist Initiate*
level 60: *Expert Mechanist*
level 70: *Arcanoengineer*
level 80: *Senior Arcanoengineer*
level 90: *Arch Arcanoenginner*
level 99: *Grand Magnus*
level 100: *Machine God*

it could be different like if you start with mage(lvl5-30) the select the subclass(artificer) the the elevated version(arcanoenrgineer) then the last lvl ("something something god")

**Comment by saertna** [comment]
Hey @soge-king,

thanks for sharing your ideas! 
Great you come up with the idea for several attributes to level up. I'm in the middle of a design in which I'm dealing with such a possibility, to have several attributes. The current stream is to level up certain attributes which certain action areas. It's still very early in the design process, so don't want to tell to much at this stage. ;)  It will be so much more fun. :-D

I recognize you have an experience in mind in which you gamify your daily task to stay motivated. I suggest, you start designing this in a sort of a game loop and attach the rewards. I believe this can be a strong motivator to keep doing the tasks you have set yourself. First on paper to get the mechanics right, and then progress into code. 

I felt also very addicted to the thought of player classes. At the end, I believe selecting a specific class should be meaningful, and with this is influencing the structure of feedback mechanics how to react on different actions. As with personal knowledge management, there are many different ways how to do it. It's quite flexible. I would be grateful for ideas how to sort the whole wild out there in some meaningful areas which can be recognized with some sort of algorithm. I can imagine this enabling class selection, or telling how much between classes you are traveling. 

Yes, having a more personalized profile representation would be great and give more recognition and care to the experience. Would love to build it in. Do you know any resource how it is achieved already in Obsidian?

In my rough road map I'm planning for more advanced statistics which will have an influence on getting XP a lot. So stay tuned. ;)

An interesting idea with a chime on certain actions. Do you know of any Obsidian plugin doing something like playing a sound? Or in general a way of doing it?

**Reply by soge-king** [reply]
from what ive seen there are some plug ins  that do give out certain "rewards" you can add in (its called Rewarder, go figure, but yea a ding after a tasks would be nice though i recommend that it can switched off since some people might not like it). honestly i would love if it could be adjustable since different people have different "rewards" for themselves and maybe a pomodoro timer as well would be cool, you could even integrate it into some functionalities. i would love to help out in designing this plug-in but im not good enough in js (only know python and C right now) and i just started obsidian like 2-3 weeks ago.

in any case, if you want an extra set of hands i could try to help out, and figure it out along the way. also i though about it the other day, and i forgot to add it but here it goes:
create a daily todo list that affects your stats like this:
![Screenshot 2024-02-06 195714](https://github.com/saertna/obsidian-gamified-pkm/assets/29467060/7b29d131-1c84-4149-9e49-a25bf29980ab)

(mundane tasks can be set up as a flat xp, if you want to expand on the lvls you can set it up like this):
![image](https://github.com/saertna/obsidian-gamified-pkm/assets/29467060/998c4518-0ea8-41f3-a0b4-45f72e6d44d7)

so different tasts can have a simple multiplier ranging from 1-5 or you could do 3 lvls (E/M/H) -> (Easy/Medium/Hard)

and you can set each folder as a "skill" and each "skill" would be associated with said stat. each new update to the folder or note would add xp to the stat then skill. for example:
[Folder]->[Skill] -> [Stat]
so say that i want to improve my coding "skills" it would look like this:
(Main folder)Coding -> (sub-folder)Skill: Python -> INT
you can divide the xp accumulation fractionally so if i gain 10xp on python because i wrote 10 lines then int gets 2xp, so the growth of a stat grows proportional to the skill of choice. something you can also do for a consequence of a missed task is either:
A) loss of xp
B) set lists of punishments

sorry if im rambling, ive just had this idea in my head for a while and this pluging is the closest ive come across to it. also, when hitting maturity, the properties could be used to associate the note with the Stat, just by using a checkmark to indicate it.

**Reply by soge-king** [reply]
also also (im sorry if im being annoying, just wanted to say it) an option to insert an image to represent the char would be great, you could honestly set it all up in the note file that has the info. here is a general idea of what it could look like:
![image](https://github.com/saertna/obsidian-gamified-pkm/assets/29467060/1a639d9e-a5a7-436d-8d81-49bb0e0cdb58)
